subprojects/docs/src/docs/userguide/jvm/toolchains.adoc

```diff
@@ -449,3 +449,14 @@ If an <<#sec:configuring_toolchain_specifications, unconfigured>> toolchain spec
 This can be achieved by passing an empty lambda when requesting a tool: `javaToolchainService.launcherFor({})`.
 
 You can find more details on defining custom tasks in the <<more_about_tasks.adoc#more_about_tasks, Authoring tasks>> documentation.
+
+[[sec:limitations]]
+== Toolchains limitations
+
+Gradle may detect toolchains incorrectly when the build is run in a JVM/JRE compiled against `musl`, https://musl.libc.org/[an alternative implementation] of the C standard library.
+A JVM/JRE compiled against `musl` may override the `LD_LIBRARY_PATH` environment variable in order to control dynamic library resolution.
+This might affect java subprocesses launched by Gradle, resulting in unexpected behavior.
```

Comment:
```suggestion
This can influence forked java processes launched by Gradle for probing toolchains, resulting in unexpected behavior.
```

Reply:
It's not just toolchain probing, any java subprocess is affected.